micronota/bfillings/_base.py

```diff
@@ -0,0 +1,83 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2015--, micronota development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from abc import ABCMeta, abstractmethod
+
+from skbio import Sequence
+
+
+class IntervalMetadataPred(metaclass=ABCMeta):
+    @abstractmethod
+    def identify_features_on_seq(self, seq, dat, **kwargs):
+        '''Identify features on the input sequence.
+
+        Parameters
+        ----------
+        seq : ``skbio.Sequence`` object
+
+        Returns
+        -------
+        ``skbio.metadata.IntervalMetadata``
+        '''
+        if isinstance(seq, Sequence):
+            pass
+
+    @abstractmethod
+    def identify_features(self, fp, dat, **kwargs):
+        '''
+        Parameters
+        ----------
+        fp : sequence file.
+
+        Yield
+        -----
+        ``skbio.metadata.IntervalMetadata``
+            The interval metadata for each sequence.
+        '''
+
+    @abstractmethod
+    def _parse_result(self):
+        '''Parse the result from running an application.'''
+
+
+class MetadataPred(metaclass=ABCMeta):
+    @abstractmethod
+    def annotate_features_on_seq(self, seq, dat, **kwargs):
+        '''Add metadata to the input seq.
+
+        Assign the function, product, cross-reference, etc. info to the
+        input sequence.
+
+        Parameters
+        ----------
+        seq : ``skbio.Sequence`` object
+
+        Returns
+        -------
+        ``skbio.metadata``
```

Comment:
input is *a* sequence, what generator to return?

On Mon, Mar 7, 2016 at 2:43 PM, Jamie Morton <notifications@github.com>
wrote:

> In micronota/bfillings/_base.py
> <https://github.com/biocore/micronota/pull/54#discussion_r55286440>:
>
> > +
> > +class MetadataPred(metaclass=ABCMeta):
> > +    @abstractmethod
> > +    def annotate_features_on_seq(self, seq, dat, **kwargs):
> > +        '''Add metadata to the input seq.
> > +
> > +        Assign the function, product, cross-reference, etc. info to the
> > +        input sequence.
> > +
> > +        Parameters
> > +        ----------
> > +        seq : ``skbio.Sequence`` object
> > +
> > +        Returns
> > +        -------
> > +        ``skbio.metadata``
>
> Also I was thinking that passing around generators would be ideal. They
> have a low memory foot-print, since it is lazily evaluated (as far as I can
> know).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/micronota/pull/54/files#r55286440>.
>